bitseq/sequence.go

```diff
@@ -306,8 +306,16 @@ func (h *Handle) validateOrdinal(ordinal uint32) error {
 }
 
 // Destroy removes from the datastore the data belonging to this handle
-func (h *Handle) Destroy() {
-	h.deleteFromStore()
+func (h *Handle) Destroy() error {
+	for {
+		if err := h.deleteFromStore(); err != nil {
```

Comment:
You need to read from store again after a retryError or this will just loop forever here in case of a retry error